src/main/java/net/projectzombie/region_rotation/modules/RegionState.java

```diff
@@ -180,52 +191,80 @@ else if (this.isPolygonRegion() && rhs.isPolygonRegion())
         }
     }
 
+    /** Used to copy blocks from the rhs state and paste them in the local state. */
     public boolean copyFrom(final RegionState rhs,
                             final boolean pasteAir)
     {
-        return _copyPaste(this, rhs, pasteAir);
+        return _copyPaste(rhs, this, pasteAir);
     }
 
+    /** Used to copy blocks from the local state and paste them in the rhs state. */
     public boolean pasteTo(final RegionState rhs,
                            final boolean pasteAir)
     {
-        return _copyPaste(rhs, this, pasteAir);
+        return _copyPaste(this, rhs, pasteAir);
     }
 
+    /**
+     * Used to copy and paste a selected region to another selected region that is the same size.
+     * @param copyState The place the blocks are coming from.
+     * @param pasteState The place the blocks are being put in.
+     * @param pasteAir If air is to be CPed into the build.
+     * @return If the CPing was successful.
+     */
     static private boolean _copyPaste(final RegionState copyState,
                                       final RegionState pasteState,
                                       final boolean pasteAir)
     {
-        if (!copyState.canRotate(pasteState)) {
+        if (copyState == null || !copyState.canRotate(pasteState)) {
             return false;
         }
 
         Iterator<Block> copyIter = copyState.getSortedBlockIterator();
         Iterator<Block> pasteIter = pasteState.getSortedBlockIterator();
         Block copyBlk, pasteBlk;
         Material copyMat, pasteMat;
-
         while (copyIter.hasNext() && pasteIter.hasNext())
         {
             copyBlk = copyIter.next();
             copyMat = copyBlk.getType();
-            pasteBlk = copyIter.next();
+            pasteBlk = pasteIter.next();
             pasteMat = pasteBlk.getType();
 
             if (!pasteAir && pasteMat.equals(Material.AIR))
             {
                 continue;
             }
-
-            if (!copyMat.equals(pasteMat))
+            // Data will have to change when MC adds material types for wool.
+            if (!copyMat.equals(pasteMat) || !(copyBlk.getData() == pasteBlk.getData()))
             {
                 pasteBlk.setType(copyMat);
                 pasteBlk.setData(copyBlk.getData());
+                if (copyBlk.getState() instanceof Chest)
```

Comment:
Okay I am pretty darn sure I did it, but there is a 1.10.2 Spigot apparently has a sign bug. If you test it tell me if the text transfers or if it goes blank.